### PR TITLE
Clarify key ID with an example

### LINKED ISSUE #321: Type2 public key is underspecified
... we should make it clear, and provide test vectors to make sure it matches. The test vector could include public keys in PEM format and then print out the expected key ID.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #321

cc @jedisct1, @fredericjacobs for an extra set of eyes to check me here -- ASN.1 is not my forte. I went based on the implementation [here](https://github.com/cloudflare/pat-go/blob/main/x509util.go#L66-L89), which I think matches what Frank did [here](https://github.com/jedisct1/rust-blind-rsa-signatures/blob/master/src/lib.rs#L320), but I would definitely appreciate confirmation!

### CODE REVIEW

File: draft-ietf-privacypass-protocol.md [hunk 1]
```diff
@@ -652,7 +652,31 @@ computed as SHA256(encoded_key), where encoded_key is a DER-encoded
...
+   36:d=4  hl=2 l=  24 cons: SEQUENCE
+   38:d=5  hl=2 l=   9 prim: OBJECT            :mgf1
+   49:d=5  hl=2 l=  11 cons: SEQUENCE
+   51:d=6  hl=2 l=   9 prim: OBJECT            :sha384
+   62:d=3  hl=2 l=   3 cons: cont [ 2 ]
+   64:d=4  hl=2 l=   1 prim: INTEGER           :30
+   67:d=1  hl=4 l= 271 prim: BIT STRING
+```
+
+And the expected key identifier based
```
Comment (chris-wood) [inline draft-ietf-privacypass-protocol.md:679]:
```suggestion
```

**Comment by chris-wood** [comment]
Friendly bump @jedisct1 and @fredericjacobs =)

**Comment by FredericJacobs** [comment]
Do you have a test vector / example to make sure it matches what I believe?

**Comment by chris-wood** [comment]
@FredericJacobs yeah, each [issuance protocol test vector](https://ietf-wg-privacypass.github.io/base-drafts/draft-ietf-privacypass-protocol.html#name-issuance-protocol-2-blind-r) includes the encoded public key (`pkS`).

**Comment by chris-wood** [comment]
Example, taken from the current test vector:

```
$ echo 30820152303d06092a864886f70d01010a3030a00d300b0609608648016503040202a11a301806092a864886f70d010108300b0609608648016503040202a2030201300382010f003082010a0282010100cb1aed6b6a95f5b1ce013a4cfcab25b94b2e64a23034e4250a7eab43c0df3a8c12993af12b111908d4b471bec31d4b6c9ad9cdda90612a2ee903523e6de5a224d6b02f09e5c374d0cfe01d8f529c500a78a2f67908fa682b5a2b430c81eaf1af72d7b5e794fc98a3139276879757ce453b526ef9bf6ceb99979b8423b90f4461a22af37aab0cf5733f7597abe44d31c732db68a181c6cbbe607d8c0e52e0655fd9996dc584eca0be87afbcd78a337d17b1dba9e828bbd81e291317144e7ff89f55619709b096cbb9ea474cead264c2073fe49740c01f00e109106066983d21e5f83f086e2e823c879cd43cef700d2a352a9babd612d03cad02db134b7e225a5f0203010001 | xxd -r -p | openssl asn1parse -dump -inform DER
    0:d=0  hl=4 l= 338 cons: SEQUENCE          
    4:d=1  hl=2 l=  61 cons: SEQUENCE          
    6:d=2  hl=2 l=   9 prim: OBJECT            :rsassaPss
   17:d=2  hl=2 l=  48 cons: SEQUENCE          
   19:d=3  hl=2 l=  13 cons: cont [ 0 ]        
   21:d=4  hl=2 l=  11 cons: SEQUENCE          
   23:d=5  hl=2 l=   9 prim: OBJECT            :sha384
   34:d=3  hl=2 l=  26 cons: cont [ 1 ]        
   36:d=4  hl=2 l=  24 cons: SEQUENCE          
   38:d=5  hl=2 l=   9 prim: OBJECT            :mgf1
   49:d=5  hl=2 l=  11 cons: SEQUENCE          
   51:d=6  hl=2 l=   9 prim: OBJECT            :sha384
   62:d=3  hl=2 l=   3 cons: cont [ 2 ]        
   64:d=4  hl=2 l=   1 prim: INTEGER           :30
   67:d=1  hl=4 l= 271 prim: BIT STRING        
      0000 - 00 30 82 01 0a 02 82 01-01 00 cb 1a ed 6b 6a 95   .0...........kj.
      0010 - f5 b1 ce 01 3a 4c fc ab-25 b9 4b 2e 64 a2 30 34   ....:L..%.K.d.04
      0020 - e4 25 0a 7e ab 43 c0 df-3a 8c 12 99 3a f1 2b 11   .%.~.C..:...:.+.
      0030 - 19 08 d4 b4 71 be c3 1d-4b 6c 9a d9 cd da 90 61   ....q...Kl.....a
      0040 - 2a 2e e9 03 52 3e 6d e5-a2 24 d6 b0 2f 09 e5 c3   *...R>m..$../...
      0050 - 74 d0 cf e0 1d 8f 52 9c-50 0a 78 a2 f6 79 08 fa   t.....R.P.x..y..
      0060 - 68 2b 5a 2b 43 0c 81 ea-f1 af 72 d7 b5 e7 94 fc   h+Z+C.....r.....
      0070 - 98 a3 13 92 76 87 97 57-ce 45 3b 52 6e f9 bf 6c   ....v..W.E;Rn..l
      0080 - eb 99 97 9b 84 23 b9 0f-44 61 a2 2a f3 7a ab 0c   .....#..Da.*.z..
      0090 - f5 73 3f 75 97 ab e4 4d-31 c7 32 db 68 a1 81 c6   .s?u...M1.2.h...
      00a0 - cb be 60 7d 8c 0e 52 e0-65 5f d9 99 6d c5 84 ec   ..`}..R.e_..m...
      00b0 - a0 be 87 af bc d7 8a 33-7d 17 b1 db a9 e8 28 bb   .......3}.....(.
      00c0 - d8 1e 29 13 17 14 4e 7f-f8 9f 55 61 97 09 b0 96   ..)...N...Ua....
      00d0 - cb b9 ea 47 4c ea d2 64-c2 07 3f e4 97 40 c0 1f   ...GL..d..?..@..
      00e0 - 00 e1 09 10 60 66 98 3d-21 e5 f8 3f 08 6e 2e 82   ....`f.=!..?.n..
      00f0 - 3c 87 9c d4 3c ef 70 0d-2a 35 2a 9b ab d6 12 d0   <...<.p.*5*.....
      0100 - 3c ad 02 db 13 4b 7e 22-5a 5f 02 03 01 00 01      <....K~"Z_.....
```

**Comment by jedisct1** [comment]
Looks good to me. But yeah, ASN.1 is not the most concise nor readable format.

**Comment by smhendrickson** [comment]
ASN1 matches my expectations as well.

**Pull request merged**